src/test/java/usecases/comment_as_guest/CagInteractorTests.java

```diff
@@ -0,0 +1,248 @@
+package usecases.comment_as_guest;
+
+import entities.comment_checkers.CommentChecker;
+import entities.display_name_checkers.DisplayNameChecker;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import usecases.RepoRes;
+import usecases.Response;
+import usecases.ThreadRegister;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class CagInteractorTests {
+    private static final ThreadRegister register = new ThreadRegister();
+
+    CagOutputBoundary pres;
+    CagGatewayComments repo;
+    CommentChecker commentChecker;
+    DisplayNameChecker displayChecker;
+
+
+    /**
+     * Customizable class to imitate CagPresenter during testing
+     */
+    static class CustomizableCagOutputBoundary implements CagOutputBoundary {
+
+        private CagOutputData receivedData;
+
+        public CustomizableCagOutputBoundary() {
+            this.receivedData = null;
+        }
+
+        public void commentAsGuestOutput(CagOutputData d) {
+            this.receivedData = d;
+        }
+
+        public CagOutputData getReceivedData() {
+            return this.receivedData;
+        }
+
+        @Override
+        public void outputShutdownServer() {
+            throw new RuntimeException("This method is not implemented and should not be called");
+        }
+    }
+
+    /**
+     * Customizable class to imitate repository with stories during testing.
+     * Records received data to be compared with applied requests
+     */
+    static class CustomizableCagGatewayComments implements CagGatewayComments {
+
+        private final boolean commentAsGuestSuccess;
+
+        /**
+         * Constructor of our customizable gateway
+         * @param commentAsGuestSuccess determines return value for commentAsGuest
+         */
+        public CustomizableCagGatewayComments(boolean commentAsGuestSuccess) {
+            this.commentAsGuestSuccess = commentAsGuestSuccess;
+        }
+
+        /**
+         * Method imitating repository's commentAsGuest with custom element based on
+         * commentAsGuestSuccess parameter.
+         * @param storyId requested storyId
+         * @param displayName proposed displayName of a guest
+         * @param comment proposed comment
+         * @return "success" when commentAsGuestSuccess is true, and "failure" otherwise
```

Comment:
i think this was originally implemented as method returning a string, so you would probably have to change it to 'return success or failure response' or something along those lines